### PR TITLE
Ft hotel models 175930848

### PR DESCRIPTION
**Pull request by leny62**
****What does this PR do?****
Creates hotel models

**Description of Task to be completed?**
-creates hotel models, that a user will be able to perform CRUD operations with hotels.
 ..a user requests a list of hotels.
..a user gets different details of hotels
..one is able to get, delete and update hotels.

**How should this be manually tested?**
1.Clone the repository/ Pull this branch
2.In your home directory move into the cloned folder by cd hotel-models
3.Then run npm i to install all dependencies.
4.Modify the connection string to database by creating a .env file, Refer to data in .env.example file.
5.Run npm start to start the server.
6. Run npm test/ npm run test to show
7.test the routes via postman
**Any background context you want to provide?**
No authentication required at the moment.

**What are the relevant Pivotal Tracker stories?**
[#175930848](https://www.pivotaltracker.com/n/projects/2476678/stories/175930848)

**Screenshots (if appropriate)**
Screen Shot 2021-01-09 at 23 24 04
Screen Shot 2021-01-09 at 23 24 18

### CODE REVIEW

**Review by kabano89 — APPROVED** [review]

**Review by furebo — CHANGES_REQUESTED** [review]
Good work , but Please add authentication to protect your endpoints

**Comment by leny62** [comment]
> Good work , but Please add authentication to protect your endpoints

**Comment by leny62** [comment]
AUthentication is now added for protection.

**Review by furebo — APPROVED** [review]

**Review by jemmyarina — CHANGES_REQUESTED** [review]
Can you add swagger documentation for your endpoints?

**Review by jemmyarina — APPROVED** [review]

**Review by Anguandia — CHANGES_REQUESTED** [review]
Please rebase and resolve conflicts

File: src/helper/signToken.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      {
+        email, password
+      },
+      JWTKEY,
+      { expiresIn: '1d' }
+    );
+  } catch (error) {
+    throw new Error('No token');
+  }
+};
```
Comment (SimonImanigirimpuhwe) [inline src/helper/signToken.js]: why this file since you don't have anywhere you are using it?

File: src/middlewares/check-auth.js [hunk 2]
```diff
@@ -8,8 +8,10 @@ module.exports = (req, res, next) => {
         req.userData = decoded;
         next();
     } catch (error) {
+        console.log('@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@', error);
+
```
Comment (SimonImanigirimpuhwe) [inline src/middlewares/check-auth.js]: Please remove unnecessary console.log

File: src/services/hotels.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    return hotels.findOne({
+      where: {
+        id: 1,
+      },
+      include: ['hotelsCity']
+    });
+  }
+
+  // static getUsers() {
+  //   return users.findAll();
```
Comment (SimonImanigirimpuhwe) [inline src/services/hotels.js]: Please consider removing files you inserted which you are no longer using and consider also removing unnecessary comments

File: src/tests/hotelModels.test.js [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+          images: ['www.unsplash.com/umubavu', 'www.gettyimages/umubavuhotel'],
+          hotelemail: 'five@yahoo.com'
+        })
+        .end((err, res) => {
+          expect(res.status).to.equal(201);
+          console.log('===========$$$$$$$', res.body.hotel.hotelId);
+          hotelId = res.body.hotel.hotelId;
+          done();
+        });
+      console.log('@@@@@@@@******@@@@@@@@@*****@@@@@@', token);
```
Comment (SimonImanigirimpuhwe) [inline src/tests/hotelModels.test.js]: After debugging consider removing these console.logs

**Review by SimonImanigirimpuhwe — APPROVED** [review]

**Review by SimonImanigirimpuhwe — CHANGES_REQUESTED** [review]
consider responding to changes.. anyways you did great!

File: src/helper/signToken.js (hunk 1 above)
Comment (leny62) [reply src/helper/signToken.js]: I will have to delete it.

File: src/services/hotels.js (hunk 3 above)
Comment (leny62) [reply src/services/hotels.js]: Sure!

File: src/middlewares/check-auth.js (hunk 2 above)
Comment (leny62) [reply src/middlewares/check-auth.js]: Yes!

**Review by SimonImanigirimpuhwe — APPROVED** [review]

**Review by Anguandia — APPROVED** [review]

**Pull request merged**